ksqldb-engine/src/main/java/io/confluent/ksql/function/udaf/attr/Attr.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udaf.attr;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.ksql.function.udaf.TableUdaf;
+import io.confluent.ksql.function.udaf.UdafDescription;
+import io.confluent.ksql.function.udaf.UdafFactory;
+import io.confluent.ksql.schema.ksql.SchemaConverters;
+import io.confluent.ksql.schema.ksql.SqlArgument;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.schema.ksql.types.SqlTypes;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.Struct;
+
+@UdafDescription(
+    name = "ATTR",
+    description = "The ATTR() aggregation indicates there are multiple values, but only one "
+        + "was expected. For example, if aggregating against a table that semantically should "
+        + "have only one value for a column given a key, this aggregation enables users to "
+        + "indicate that they expect only a single value.\n\n"
+        + "If the aggregation encounters more than a single value for the expected singular "
+        + "column, the entire aggregation will return null."
```

Comment:
Any particular reason we are returning null instead of something like a "*" as Tableau does? What if all the column values before aggregation were actually null?

Reply:
`"*"` is a string value, which is not applicable to all value formats. if all columns are actually null then the aggregation will (correctly) return null (fwiw, this comment is applicable for all defaults, including something like `*`).